lib/ui/window/platform_configuration.cc

```diff
@@ -121,26 +121,29 @@ void PlatformConfiguration::AddView(int64_t view_id,
       }));
 }
 
-void PlatformConfiguration::RemoveView(int64_t view_id) {
+bool PlatformConfiguration::RemoveView(int64_t view_id) {
   if (view_id == kFlutterImplicitViewId) {
     FML_LOG(ERROR) << "The implicit view #" << view_id << " cannot be removed.";
-    FML_DCHECK(false);
-    return;
+    FML_UNREACHABLE();
```

Comment:
Perhaps I'm misreading the code, but this doesn't seem like it's so much _unreachable_ as it is a violation of the contract of this method. In other words this method itself is callable, but someone's calling it with a value that we claim is not allowed. I'd generally reserve `FML_UNREACHABLE` for scenarios like after a switch statement that covers all values and is meant to return in all cases -- i.e. situations that should be "impossible". 

If this method is only other called from other engine code, then this is probably fine as-is, though I'd still be tempted to just replace the `FML_LOG(ERROR)` above with an `FML_LOG(FATAL)` and leave it at that.

Reply:
Excellent point, will update!